### DISCUSSION TITLE
Add max-file to the logging driver specific options

### DISCUSSION THREAD

**Discussion by aklira** [containers/podman#20985] [Ideas]
In the current version of Podman, the only option available for configuring log files size is the "max-size," which sets the maximum size of the log file. However, a limitation arises when the file is rotated, as only the latest traces are accessible, and the complete history is lost.

To address this issue, it would be highly beneficial to introduce a "max-file" option within the logging driver specific options. This option would allow users to specify the maximum number of log files that can coexist. In the event of log rotation creating surplus files, the oldest file would be automatically removed.

For instance, if the user sets "--log-opt max-file=3," Podman would generate a maximum of three files to store the current log information. Each file would adhere to the specified maximum size set by "--log-opt max-size," ensuring a more comprehensive and organized log history.

### Example:
traces.log, (current file)
traces.log.1,
traces.log.2
traces.log.3

**Comment by AhmadKasem** [comment]
This option is available in podman and it works, it's just not documented for some reason.

**Reply by rhatdan** [reply]
Which option?

**Reply by AhmadKasem** [reply]
> Which option?

I was talking about the --log-opt max-file option. But after some digging, I'm not sure it works and I need to test it.

**Reply by rhatdan** [reply]
There is nothing in the code that handles max-file, that I can find.

**Comment by cfeltz34** [comment]
I tested the following command:

```
podman run -d --rm \
  --log-driver=k8s-file \
  --log-opt path=/var/log/k8s_files/podman/log-tester.json \
  --log-opt max-size=10k \
  --log-opt max-files=5 \
  --name log-tester \
  alpine sh -c "while true; do echo 'Generating log entry' \$(date); sleep 0.1; done"

```

In this setup, even though I specified --log-opt max-files=5, it seems that the k8s-file driver ignores this option, only utilizing max-size. Therefore, I lose the ability to manage the number of log files created, which can lead to issues with log storage management.

Introducing the max-file option would be highly beneficial, as it would allow us to maintain a more organized log history by automatically removing the oldest log files when the limit is reached.

I look forward to your insights on this matter. Thank you!

**Reply by LennertMertens** [reply]
I'm running into the same issue. In my opinion the max-size is quite pointless if you have no option to retain older logs via proper log rotation. I'm looking into working around this issue with logrotate but I would be glad if it's just supported in Podman as well.
